Tools/TestWebKitAPI/Tests/WebKitCocoa/LoadAndDecodeImage.mm

```diff
@@ -266,10 +268,30 @@
     Util::run(&done);
 }
 
+RetainPtr<NSData> tiffRepresentation(CocoaImage *image)
+{
+#if USE(APPKIT)
+    return [image TIFFRepresentation];
+#else
+    RetainPtr cgImage = [image CGImage];
+    if (!cgImage)
+        return nullptr;
+
+    RefPtr bitmapImage = WebCore::BitmapImage::create(WTFMove(cgImage));
+    if (!bitmapImage)
+        return nullptr;
+
+    WebCore::ImageAdapter adapter { *bitmapImage };
+    return bridge_cast(adapter.tiffRepresentation());
```

Comment:
I don't think you need to create a `BitmapImage` or `ImageAdapter` here. You can use the static method `ImageAdapter::tiffRepresentation()`
```
WebCore::ImageAdapter::tiffRepresentation({ WebCore::NativeImage::create(WTFMove(cgImage)) });
```